File: internal/db/models/utilities/cloud-init.go
package utilitiesModels

import "time"

type CloudInitTemplate struct {
	ID   uint   `gorm:"primaryKey" json:"id"`
	Name string `json:"name" gorm:"uniqueIndex;"`
	User string `json:"user"`
	Meta string `json:"meta"`
	NetworkConfig string `json:"networkConfig"`

	CreatedAt time.Time `json:"createdAt" gorm:"autoCreateTime"`
	UpdatedAt time.Time `json:"updatedAt" gorm:"autoUpdateTime"`
}

File: web/src/lib/types/utilities/cloud-init.ts
import z from "zod/v4";

export const CloudInitTemplateSchema = z.object({
    id: z.number().int().nonnegative(),
    name: z.string(),
    user: z.string(),
    meta: z.string(),
    networkConfig: z.string(),
    createdAt: z.string(),
    updatedAt: z.string()
});

export type CloudInitTemplate = z.infer<typeof CloudInitTemplateSchema>;